frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/CheckListPopup.tsx

```diff
@@ -0,0 +1,179 @@
+    if (e.key === 'Enter') {
+      void onSaveNewOptionClick(index);
+    }
+  };
+
+  const onSaveNewOptionClick = async (index: number) => {
+    await new SelectOptionCellBackendService(cellIdentifier).createOption({ name: newOptions[index] });
+    setNewOptions(newOptions.filter((_, i) => i !== index));
+  };
+
+  return (
+    <PopupWindow className={'text-xs'} onOutsideClick={onOutsideClick} left={left} top={top}>
+      <div className={'min-w-[320px]'}>
+        <div className={'px-4 pt-8 pb-4'}>
+          <CheckListProgress completed={selectedOptionsCount} max={allOptionsCount} />
+        </div>
+
+        <div className={'flex flex-col p-2'}>
+          {(databaseStore.fields[cellIdentifier.fieldId]?.fieldOptions as ISelectOptionType).selectOptions.map(
+            (option, index) => (
+              <div
+                key={index}
+                className={
+                  'flex cursor-pointer items-center justify-between rounded-lg px-2 py-1.5 hover:bg-main-secondary'
+                }
+                onClick={() =>
+                  onToggleOptionClick(
+                    new SelectOptionPB({
+                      id: option.selectOptionId,
+                      name: option.title,
+                      color: option.color || SelectOptionColorPB.Purple,
+                    })
+                  )
+                }
+              >
+                <div className={'h-5 w-5'}>
+                  {(data as SelectOptionCellDataPB)?.select_options?.find((so) => so.id === option.selectOptionId) ? (
+                    <EditorCheckSvg></EditorCheckSvg>
+                  ) : (
+                    <EditorUncheckSvg></EditorUncheckSvg>
+                  )}
+                </div>
+                <div className={`flex-1 px-2 py-0.5`}>{option.title}</div>
+                <div className={'flex items-center'}>
+                  <button onClick={(e) => onCheckListDetailClick(e, option)} className={'h-6 w-6 p-1'}>
+                    <Details2Svg></Details2Svg>
+                  </button>
+                </div>
+              </div>
+            )
+          )}
+          {newOptions.map((option, index) => (
+            <div
+              key={index}
+              className={'flex cursor-pointer items-center justify-between rounded-lg px-2 py-1.5 hover:bg-shade-6'}
+            >
+              <input
+                onKeyDown={(e) => onNewOptionKeyDown(index, e as unknown as KeyboardEvent)}
+                className={'min-w-0 flex-1 pl-7'}
+                value={option}
+                onChange={(e) => updateNewOption(index, e.target.value)}
+              />
+              <button
+                onClick={() => onSaveNewOptionClick(index)}
+                className={
+                  'flex items-center gap-2 rounded-lg bg-main-accent px-4 py-2 text-white hover:bg-main-hovered'
+                }
+              >
+                {t('grid.selectOption.create')}
+              </button>
+            </div>
```

Comment:
`<OptionInput />`

Reply:
did some refactoring and reorganizing for multiselect and checklist field types